src/main/java/org/folio/circulation/domain/policy/library/EndOfCurrentHoursStrategy.java

```diff
@@ -23,9 +25,37 @@ public EndOfCurrentHoursStrategy(DateTime currentTime, DateTimeZone zone) {
     if (currentTimeInterval == null) {
       return failed(failureForAbsentTimetable());
     }
+    if(hasLibraryRolloverWorkingDay(libraryTimetable, requestedInterval)){
+      return succeeded(requestedInterval.getPrevious().getEndTime());
+    }
     if (currentTimeInterval.isOpen()) {
       return succeeded(currentTimeInterval.getEndTime());
     }
     return succeeded(currentTimeInterval.getNext().getEndTime());
   }
+
+  public static final LocalTime END_OF_A_DAY = LocalTime.MIDNIGHT.minusSeconds(1);
+
+  private boolean hasLibraryRolloverWorkingDay(LibraryTimetable libraryTimetable,
+                                               LibraryInterval requestedInterval) {
+
+    if(isNotSequenceOfWorkingDays(libraryTimetable, requestedInterval)) {
+      return false;
+    }
+
+    LocalTime endLocalTime = libraryTimetable.getHead().getEndTime().toLocalTime();
+    LocalTime startLocalTime = requestedInterval.getPrevious().getStartTime().toLocalTime();
+
+    return isDateEqualToRange(endLocalTime, LocalTime.MIDNIGHT.minusMinutes(1))
+      && isDateEqualToRange(startLocalTime, LocalTime.MIDNIGHT);
+  }
+
+  private boolean isNotSequenceOfWorkingDays(LibraryTimetable libraryTimetable, LibraryInterval requestedInterval) {
+    return Days.daysBetween(libraryTimetable.getHead().getEndTime(),
+      requestedInterval.getPrevious().getStartTime()) != Days.ZERO;
+  }
+
+  private boolean isDateEqualToRange(LocalTime requestedInterval, LocalTime range) {
```

Comment:
How can a date be equal to a range?

The second parameter appears to also be a date, rather than a range, which means this is comparing two dates. How is a range involved?

Reply:
This method checks that working time of library equals the end of day (23:59) or the begin of day(00:00). This is second scenario for rollover library working hours when the library works till 23.59 in current day end and it continues working on the next day from 00:00 to some time. I updated the name of the method.